nat-lab/tests/test_upnp_connection.py

```diff
@@ -264,7 +264,7 @@ async def test_upnp_port_lease_duration(
             await temp_exit_stack.enter_async_context(alpha_gw_router.reset_upnpd())
 
         # this check should be done before endpoint interval triggers a new mapping (optimal > 5s)
-        upnpc_cmd = await execute_upnpc_with_retry(alpha_conn)
+        upnpc_cmd = await alpha_conn.create_process(["upnpc", "-i", "-l"]).execute()
```

Comment:
why reverting back from `execute_upnpc_with_retry`. Won't we just hit previous issue we had again?

Reply:
Or maybe here we should get specific exit status?

Reply:
This should check if the lease does _does not_ exists (hence `assert re.search("^ [0-9]+ UDP", ....) is None`). So we shouldn't check it for 10 seconds. The `execute_upnpc_with_retry` to repeat, if we actually want to find the lease.

Reply:
ok, but won't this call throw some sort of `ProcessExecError` and then fail the test if `upnpd` is not started yet?

Reply:
In case of not working `upnpd` - yes it _should_ fail, but than the test should fail, because we're expecting to be running, only those leases should be expired.